build failure with giza_box_time, causing build failure in macports giza

from: https://build.macports.org/builders/ports-10.5_ppc_legacy-builder/builds/80835/steps/install-port/logs/stdio

```/bin/sh ../libtool  --tag=CC   --mode=compile /usr/bin/gcc-4.2 -DHAVE_CONFIG_H -I. -I..   -I/opt/local/include/cairo -I/opt/local/include/glib-2.0 -I/opt/local/lib/glib-2.0/include -I/opt/local/include/pixman-1 -I/opt/local/include/ossp -I/opt/local/include/freetype2 -I/opt/local/include/libpng16 -I/opt/local/include  -pipe -Os -arch ppc -MT libgiza_la-giza-box-time.lo -MD -MP -MF .deps/libgiza_la-giza-box-time.Tpo -c -o libgiza_la-giza-box-time.lo `test -f 'giza-box-time.c' || echo './'`giza-box-time.c
libtool: compile:  /usr/bin/gcc-4.2 -DHAVE_CONFIG_H -I. -I.. -I/opt/local/include/cairo -I/opt/local/include/glib-2.0 -I/opt/local/lib/glib-2.0/include -I/opt/local/include/pixman-1 -I/opt/local/include/ossp -I/opt/local/include/freetype2 -I/opt/local/include/libpng16 -I/opt/local/include -pipe -Os -arch ppc -MT libgiza_la-giza-box-time.lo -MD -MP -MF .deps/libgiza_la-giza-box-time.Tpo -c giza-box-time.c  -fno-common -DPIC -o .libs/libgiza_la-giza-box-time.o
giza-box-time.c: In function ‘giza_box_time’:
giza-box-time.c:137: error: ‘for’ loop initial declaration used outside C99 mode
giza-box-time.c:139: error: redefinition of ‘p’
giza-box-time.c:137: error: previous definition of ‘p’ was here
giza-box-time.c:139: error: ‘for’ loop initial declaration used outside C99 mode
giza-box-time.c:178: error: redeclaration of ‘p’ with no linkage
giza-box-time.c:139: error: previous definition of ‘p’ was here
make[2]: *** [libgiza_la-giza-box-time.lo] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_science_giza/giza/work/giza-1.0.0/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_science_giza/giza/work/giza-1.0.0'
make: *** [all] Error 2```

--- a/src/giza-box-time.c
+++ b/src/giza-box-time.c
@@ -134,10 +134,10 @@ int pgnpl(int nmax, int n) {
   }
   /* Now that we have a copy of the strings we can easily upcase them -
      makes parsing easier */
-  for(char* p = xopt_cp; *p; p++ )
-      *p = toupper(*p);
-  for(char* p = yopt_cp; *p; p++ )
-      *p = toupper(*p);
+  for(char* px = xopt_cp; *px; px++ )
+      *px = toupper(*px);
+  for(char* py = yopt_cp; *py; py++ )
+      *py = toupper(*py);
 
   /* X-axis */
   if( strchr(xopt_cp, 'Z') ) {

